parser.c

```diff
@@ -33,6 +33,7 @@ static int parser_read_char(struct kanshi_parser *parser) {
 		return ch;
 	}
 
+	errno = 0;
```

Comment:
This is unnecessary, `errno` is only checked when `fgetc` errors out, in which case it's guaranteed to be set to `fgetc's` error.